### PR TITLE
1340 New page for Guilds sign up

### LINKED ISSUE #1340: New guilds conversion/signup page
Need someone to build the new guilds conversion/signup page so we can evolve from the [current one](https://my.metagame.wtf/join/guild) with an ugly block of text into a [new, prettier & more enticing version](https://www.figma.com/proto/n36ggiRBQEGzRUbGxWfkMX/Meta-Game-(Copy)?node-id=5853%3A4465&scaling=scale-down&page-id=5836%3A3265).

If anyone is interested in working on it, pls comment here & I'll assign you.

Infographic exported as an image to make it easier:
Group 1005

### PR DESCRIPTION
**Pull request by HHH-GH**
## Overview

**What features/fixes does this PR include?**

Implements fiyin and peth's [new design and content for the Guilds Join page](https://www.figma.com/file/n36ggiRBQEGzRUbGxWfkMX/Meta%20Game%20(Copy)?node-id=5853%3A4465&fuid=1118498784933426378)

(Maybe not perfectly)

**Please provide the GitHub issue number**

Closes #1340 

## Implementation
- Components for each section of the page are loaded into `packages/web/pages/join/guild/index.tsx` from `packages/web/components/Guild/Join/` (except for the list of guilds)

**Describe technical (nontrivial / non-obvious) parts of your code**
- n/a, all standard Chakra UI components for the layout. There are some responsive behaviours, and I wrote some comments to explain those

### Difficulties encountered
1. The design called for a grid/card list of six current guilds, with a LOAD MORE button. This part was beyond my skillz. ~~, in particular the writing of a query to get six guilds and load in more on the click. What I have done is grabbed the entire list of guilds (as per `packages/web/components/GuildList.tsx`), and put them in a grid *after* the APPLY button and CTA (because having the full list means way to much scrolling to get to the APPLY button).~~  
UPDATE: What I have done is limited the getGuilds query to 6 guilds and put a View More Guilds button in place of a Load More button
2. ~~I also couldn't figure out how to write it so the GuildList could be pulled in from a component, so it's inlined in the `packages/web/pages/join/guild/index.tsx`~~  
UPDATE: The code for the section with the list of six guilds is inlined in `packages/web/pages/join/guild/index.tsx` instead of imported from a component because `index.tsx` is a static page and `getStaticProps` won’t work in a component imported into a static page.

### Other notes
- I freestyled the design for md/lg breakpoints
- Nothing special really, based mostly on existing code and easy Chakra UI components.
- ~~Would appreciate any help with the Guild List part~~
UPDATE: the Guild List part is fine as-is, I think.

## See the preview?
Link: https://frontend-pr-1353-mjhnbmqqna-uk.a.run.app/join/guild

## Assets
Added assets to the `web/assets` folder, see screenshot

image

### CODE REVIEW

**Comment by HHH-GH** [comment]
I updated some more things  

- The list of guilds matches fiyin's design, except instead of being able to load more guilds into the page you can click a button to go to the existing view all the guilds page
- The background of the cards/tiles/whatevers match the rest of the site (rounded corners, alpha transparency, blur, box shadow)
- I made some CSS styles to control the background of the cards/tiles/whatevers so I can update them from one place (if needed)

**Comment by HHH-GH** [comment]
Finished the last set of text updates and tweaks requested by @peth-yursick 

Ready for review! 🎉

**Review by alalonde — APPROVED** [review]
Pushing this through anyway; we definitely need to revisit the guild signup flow still

File: packages/web/components/Guild/Join/Requirements.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+              Patrons are the ones <Text as="em">passively</Text> contributing
+              towards the realization of MetaGame.
+            </Text>
+            <Box>
+              <MetaButton
+                as="a"
+                bg="#E839B7"
+                borderRadius={0}
+                color="white"
+                href="https://metagame.wtf/community/join/patrons"
```
[the remark this answers is not in the document]
Comment (HHH-GH) [reply packages/web/components/Guild/Join/Requirements.tsx:138]: Noted for next time, and let me know if you still want me to clean up those links

**Pull request merged**